### ISSUE TITLE
Switch to Saml2 (Azure AD) as identity provider

### ISSUE THREAD

**Issue by amercader** [okfn/ckanext-unhcr#193]
This essentially means switching from ckanext-ldap to ckanext-saml2.

ckanext-saml2 is not the cleanest extension and takes a while to navigate all the cnfiguration instructions but essentially what we need on each site is a federation metadata XML file (`idp.xml`) and a python config file (`sp_config.py`). On the gates project the saml2 extension has been modified to support CKAN 2.8 and other features but let's start with the version at https://github.com/okfn/ckanext-saml2 and see if we need something else.

I asked the UNCRH people to set up an Azure AD app for UAT and one for our local dev env, which should make things much easier for us.
- [x] Request files needed to UNCHR
- [x] Install ckanext-saml2 and requirements in image. Remove LDAP ones.
- [x] Follow the ckanext-saml2 README instructions to configure it. Specially important are the config files and the mapping of fields between the Saml2 response and CKAN (`saml2.user_mapping`)
- [x] If we get an application for localhost:5000, that will need to be served using https. We can use a self signed cert for paster. I'll share the details when we start work
- [x] Test login locally
- [x] Deploy and test on UAT

**Comment by amercader** [comment]
To serve the application under https in the dev server (https://locahost:5000) we need to modify the .ini file as such:

```
[server:main]
use = egg:Paste#http
host = 0.0.0.0
port = 5000
ssl_pem = host.pem

ckan.site_url = https://localhost:5000

```

host.pem is just a self signed certificate. I'll DM a copy with the appropriate format.

**Comment by amercader** [comment]
I'll share the `idp.xml` file to be used privately as well

**Comment by roll** [comment]
I've bootstrapped the migrations here:
- https://github.com/okfn/docker-ckan-unhcr/pull/12 
- https://github.com/okfn/ckanext-unhcr/pull/207

There are a lot of outstanding issues I don't understand or worry about:
- the step `Generate the sp metadata (sp.xml): /usr/lib/ckan/bin/python /usr/lib/ckan/src/pysaml2/tools/make_metadata.py /usr/lib/ckan/src/ckanext-saml2/ckanext/saml2/config/sp_config.py > sp.xml (the paths to python, make_metadata.py sp_config.py might vary depending on where you installed ckan in your virtual env)` doesn't really say what to do with this `sp.xml` file and what is it. In the reference project (G) I wasn't able to find its generation.
- it's not clear where to get `mykey.pem`. I just used the one from `ckanext-saml2`, for now. But it's not correct obviously
- because we use `Alpine` Linux I'm not sure we can install all the required dependencies from the package repository. I added ones that I found and was able to install.
- a lot of other stuff are not clear from the installation instruction like how to change `sp_config.py`, `who.ini` etc

I hope my first steps will help but we're parking it, for now, while Adria is away. For now, visiting the `user/login` page fails with `Ignoring REMOTE_USER - does not look like a NameID` and a redirects loop
